### ISSUE TITLE
tmux/osx + ttymouse=sgr

### ISSUE THREAD

**Issue by cvlmtg** [neovim/neovim#1862]
If I set "ttymouse=sgr", neovim inserts random characters every time I scroll with the mouse wheel. leaving ttymouse to the standard "xterm" does not give any problems, but I can't click to place the cursor beyond colum 223 (a known limitation)

this happens only when nvim runs under tmux

**Comment by fwalch** [comment]
Related: #1845.

**Comment by Pyrohh** [comment]
Is this still an issue when using #1820?

**Comment by cvlmtg** [comment]
I've checked out tarruda's tickit-ui branch and compiled it. scroll under tmux works now. just a couple of things:

* loading my .nvimrc crashes neovim (which doesn't happen with the current master branch)
* if I set ttymouse=sgr I have the random characters again :) so yes, after merging this new branch it might be a good idea to remove the ttymouse setting to avoid problems or confusing users

**Comment by cvlmtg** [comment]
I found something else quite strange...

```
 (tickit-ui) ~/d/neovim build/bin/nvim -u NONE
fish: Job 1, 'build/bin/nvim -u NONE' terminated by signal SIGSEGV (Address boundary error)
 (tickit-ui) ~/d/neovim cd build/bin/
 (tickit-ui) ~/d/n/b/bin nvim -u NONE
```

the last command works. so it seems neovim runs only from the current directory :)

**Comment by fwalch** [comment]
> If I click with the mouse beyond column 223 the cursor doesn't move.

Hm, seems to have worked in https://github.com/neovim/neovim/issues/1845#issuecomment-70988090. (cc @tarruda)

**Comment by cvlmtg** [comment]
sorry, I should never do any test before my coffee :) I still had neovim installed with homebrew and I think I run it by mistake instead of the one I compiled from tarruda branch, so I was actually using one binary while believing I was using the other one... so here's the correct report :)

tickit-ui version of vim *without tmux* works ok and I can place the cursor beyond column 223

*inside tmux* this is what happens:

```
 (tickit-ui) ~/d/neovim build/bin/nvim
No neovim module found for Python 2.7.6. Try installing it with 'pip install neovim' or see ':help nvim-python'.
Press ENTER or type command to continue
fish: Job 1, 'build/bin/nvim' terminated by signal SIGSEGV (Address boundary error)
 (tickit-ui) ~/d/neovim build/bin/nvim -u NONE
fish: Job 1, 'build/bin/nvim -u NONE' terminated by signal SIGSEGV (Address boundary error)
 (tickit-ui) ~/d/neovim cd build/bin/
 (tickit-ui) ~/d/n/b/bin ./nvim
No neovim module found for Python 2.7.6. Try installing it with 'pip install neovim' or see ':help nvim-python'.
Press ENTER or type command to continue
fish: Job 1, './nvim ' terminated by signal SIGSEGV (Address boundary error)
 (tickit-ui) ~/d/n/b/bin ./nvim -u NONE
fish: Job 1, './nvim -u NONE' terminated by signal SIGSEGV (Address boundary error)
 (tickit-ui) ~/d/n/b/bin
```

I'm running on OSX 10.10, homebrew tmux and iterm2

**Comment by Pyrohh** [comment]
I assume this can be closed now that #1820 is merged.
